api/v1alpha1/promotion_types.go

```diff
@@ -36,6 +37,33 @@ const (
 	PromotionPhaseAborted PromotionPhase = "Aborted"
 )
 
+type PromotionStepPhase string
```

Comment:
Should we still call this a "phase" if we further refer to it as a "status"?

Reply:
I was only mirroring the existing `PromotionPhase` type (which may also be a misnomer).

I'm not married to the name and would happily consider something different.

I had momentarily considered `PromotionStepStatus`, but felt this _sounded_ more like a struct than a string type.

Reply:
(Sounds that way to me because k8s has me conditioned to think of "status" as a compound type.)

Reply:
Spoke offline and settled on status.